src/NLog.Database/DatabaseParameterInfo.cs

```diff
@@ -48,6 +49,160 @@ namespace NLog.Targets
     [NLogConfigurationItem]
     public class DatabaseParameterInfo
     {
+        /// <summary>
+        /// Mapping of System.Data.DbType, System.Data.SqlDbType and NpgsqlTypes.NpgsqlDbType members names
+        /// to their corresponding CLR types.
+        /// </summary>
+        private static readonly Dictionary<string, Type> _typesByDbTypeName =
+            new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase)
+            {
+                // Members from System.Data.DbType
+
+                { nameof(System.Data.DbType.AnsiString), typeof(string) },
+                // { nameof(System.Data.DbType.Binary), typeof(byte[]) },  // not supported
+                { nameof(System.Data.DbType.Byte), typeof(byte) },
+                { nameof(System.Data.DbType.Boolean), typeof(bool) },
+                { nameof(System.Data.DbType.Currency), typeof(decimal) },
+                { nameof(System.Data.DbType.Date), typeof(DateTime) }, // DateOnly when .net framework will be deprecated
+                { nameof(System.Data.DbType.DateTime), typeof(DateTime) },
+                { nameof(System.Data.DbType.Decimal), typeof(decimal) },
+                { nameof(System.Data.DbType.Double), typeof(double) },
+                { nameof(System.Data.DbType.Guid), typeof(Guid) },
+                { nameof(System.Data.DbType.Int16), typeof(short) },
+                { nameof(System.Data.DbType.Int32), typeof(int) },
+                { nameof(System.Data.DbType.Int64), typeof(long) },
+                { nameof(System.Data.DbType.Object), typeof(object) }, // not sure if we should support this
+                { nameof(System.Data.DbType.SByte), typeof(sbyte) },
+                { nameof(System.Data.DbType.Single), typeof(float) },
+                { nameof(System.Data.DbType.String), typeof(string) },
+                { nameof(System.Data.DbType.Time), typeof(TimeSpan) }, // TimeOnly when .net framework will be deprecated
+                { nameof(System.Data.DbType.UInt16), typeof(ushort) },
+                { nameof(System.Data.DbType.UInt32), typeof(uint) },
+                { nameof(System.Data.DbType.UInt64), typeof(ulong) },
+                { nameof(System.Data.DbType.VarNumeric), typeof(decimal) },
+                { nameof(System.Data.DbType.AnsiStringFixedLength), typeof(string) },
+                { nameof(System.Data.DbType.StringFixedLength), typeof(string) },
+                { nameof(System.Data.DbType.Xml), typeof(string) },
+                { nameof(System.Data.DbType.DateTime2), typeof(DateTime) },
+                { nameof(System.Data.DbType.DateTimeOffset), typeof(DateTimeOffset) },
+
+
+                // Members from System.Data.SqlDbType
+
+                { "BigInt", typeof(long) },
+                // { "Binary", typeof(byte[]) }, // not supported
+                { "Bit", typeof(bool?) },
```

Comment:
Why nullable ? Is bit tri-state by default ?

Reply:
According to [the doc](https://docs.microsoft.com/en-us/dotnet/api/system.data.sqldbtype) yes. :confused:

Reply:
I prefer to stay away from very provider specific implementations, espcially when not able to verify if the that specific DbProvider is being used. Not sure I like this direction of trying to recognizes all DbTypes independent of DbProvider. But prefer to to just to recognize the most obvious ones.

Reply:
I put `SqlDbType` and `NpgsqlDbType` because the doc states it's feasible. 

But I think that:

1. The `dbType` parameter should only accept names from `System.Data.DbType`.
2. `contains('Timestamp')` should be removed, it's ambiguous since it matches a `DateTime` for _NpgsqlDbType_ and a `byte[8]` for _SqlDbType_
3. `contains('Bool')` is replaced by `equals('Boolean')`

The last two points break the backward compatibility.

[The doc](https://github.com/NLog/NLog/wiki/Database-target) needs to be updated accordingly.  
By the way the examples are `SqlDbType.NChar` and `NpgsqlDbType.Json` and both are not parsed.

Reply:
Since the method was only "best effort" and never meant as exact match for all major dbproviders (or lesser known), then I think the docs are very vague. Again it is mostly for convience, so it is easy to setup for the normal dbtypes (not for all dbtypes).